src/components/templates/OrderTemplate.jsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import { getCart } from '../../services/cart';
+import {useQuery,useMutation} from 'react-query';
+import {useNavigate} from 'react-router-dom'
+import { comma } from '../../utils/convert';
+import { order } from '../../services/order';
+import routes from '../../routes/routes';
+import { useState } from 'react';
+import { useRef } from 'react';
+import OptionList from '../atoms/OptionList';
+
+const OrderTemplate = ({ data }) => {
+
+    const [allAgree,setAllagree]=useState(false)
+    const [agreePolicy,setAgreePolicy]=useState(false);
+    const [agreePayment,setAgreePayment]=useState(false);
+    // const [agree,setAllAgree]=useState(false);
+
+    const allAgreeRef = useRef(null);
+    const agreePaymentRef = useRef(null);
+    const agreePolicyRef =useRef(null)
+
+    const handleAllAgree=(e)=>{
+        const value= e.target.checked;
+        setAgreePolicy(value)
+        setAgreePayment(value)
+    }
+    const handleAgreement=(e)=>{
+        const {name,checked}=e.target;
+        if(name==='payment-agree'){
+            setAgreePayment(checked)
+        }else if(name==='policy-agree'){
+            setAgreePayment(checked)
+        }
+        if(!checked){
+            setAllagree(false)
+        }
+    }
+
+
+    const {products,totalPrice}=data?.data?.response;
+
+    const navigate=useNavigate();
+    const {mutation}=useMutation({
+        mutationKey:'order',
+        queryFn:order,
+    })
+    const {data:Data, error }= useQuery(getCart);
+
+    const OrderItems=()=>{
+        let renderComponent=[];
+        if(Array.isArray(products)===false) return;
```

Comment:
```suggestion
        if(!Array.isArray(products)) return;
```

위처럼 간단하게 작성할 수 있을 것 같아요.